crates/provider/src/native.rs

```diff
@@ -0,0 +1,323 @@
+use std::error::Error;
+
+use crate::shared::{
+    client::Client,
+    types::{FileMap, NameSpaceDef, PodDef, RunCommandOptions, RunCommandResponse, Settings},
+};
+
+struct Native {
+    // Namespace of the client
```

Comment:
```suggestion
    /// Namespace of the client
```
We should use `///` for [doc comments](https://doc.rust-lang.org/rust-by-example/meta/doc.html#doc-comments)